src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,296 @@
+ *  - The `quit` method allows moving outstanding GEM balance to `quitTo`. It can be called only by the admin.
+ *  - The `quit` method with `amount` argument allows moving specified amount of GEM balance to `quitTo`.
+ *  - The `file` method allows updating `quitTo`, `to` addresses. It can be called only by the admin.
+ */
+contract RwaInputConduit3 {
+    /// @notice PSM GEM token contract address
+    GemAbstract public immutable gem;
+    /// @notice PSM contract address
+    PsmAbstract public immutable psm;
+    /// @dev DAI/GEM decimal difference.
+    uint256 private immutable to18ConvertionFactor;
+
+    /// @dev This is declared here so the storage layout lines up with RwaInputConduit.
+    address private __unused_gov;
+    /// @notice Dai token contract address
+    DaiAbstract public dai;
+    /// @notice RWA urn contract address
+    address public to;
+
+    /// @notice Addresses with admin access on this contract. `wards[usr]`
+    mapping(address => uint256) public wards;
+    /// @notice Addresses with push access on this contract. `may[usr]`
+    mapping(address => uint256) public may;
+
+    /// @notice Exit address
+    address public quitTo;
+
+    /**
+     * @notice `usr` was granted admin access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` admin access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice `usr` was granted push access.
+     * @param usr The user address.
+     */
+    event Mate(address indexed usr);
+    /**
+     * @notice `usr` push access was revoked.
+     * @param usr The user address.
+     */
+    event Hate(address indexed usr);
+    /**
+     * @notice `wad` amount of Dai was pushed to `to`
+     * @param to The RwaUrn address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. Currently the supported values are: "quitTo", "to".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The conduit outstanding gem balance was flushed out to `exitAddress`.
+     * @param quitTo The quitTo address.
+     * @param wad The amount flushed out.
+     */
+    event Quit(address indexed quitTo, uint256 wad);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaInputConduit3/not-authorized");
+        _;
+    }
+
+    modifier isMate() {
+        require(may[msg.sender] == 1, "RwaInputConduit3/not-mate");
+        _;
+    }
+
+    /**
+     * @notice Define addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     * @param _to RwaUrn contract address.
+     */
+    constructor(address _psm, address _to) public {
```

Comment:
If we decide to do the `psm` fileable I might suggest to also pass `dai` and `gem` and put requires to check they match with the psm ones.

Reply:
We can still fetch `dai` and `gem` from the PSM and store them as immutable.
After that, when calling `file()`, we just check whether the new PSM `dai` and `gem` match the stored ones.

Reply:
As long as `DAI` and `gem` remain the same, we can swap the PSM instances, as it is just an implementation detail.

IMO It's a bit pointless to pass `address dai` and `address gem` in the constructor only to compare them agains `psm.gem()` and `psm.dai()`.

Reply:
I'll repeat, I'm aware that from the practical point of view is the same end goal and it is actually shorter to achieve.
However, defining in the constructor `dai` and `gem` just feels from a semantical point of view more explicitly correct as they are the constants that you want your variable (the `psm`) respects.
Anyway as I've also said, nothing I feel strongly and it is clearly super subjective.